geopandas/io/parquet.py

```diff
@@ -0,0 +1,223 @@
+import json
+import warnings
+
+from pandas.io.common import get_filepath_or_buffer
+from pandas.io.parquet import PyArrowImpl as PandasPyArrowImpl
+from pandas import DataFrame
+
+from shapely.wkb import loads
+
+from geopandas import GeoDataFrame, GeoSeries
+
+
+def _encode_crs(crs):
+    if isinstance(crs, str):
+        return {"proj4": crs}
+
+    return crs
+
+
+def _decode_crs(crs):
+    if crs and "proj4" in crs:
+        return crs["proj4"]
+
+    return crs
+
+
+class PyArrowImpl(PandasPyArrowImpl):
+    """Extension of Pandas PyArrowImpl to handle serializing geometries to WKB.
+    """
+
+    def write(
+        self,
+        df,
+        path,
+        compression="snappy",
+        coerce_timestamps="ms",
+        index=None,
+        **kwargs
+    ):
+        """Write data frame to a parquet file, serializing any geometry
+        columns to WKB format.
+
+        See docstring for ``to_parquet``.
+        """
+
+        self.validate_dataframe(df)
+
+        geometry_columns = df.dtypes.loc[df.dtypes == "geometry"].index.tolist()
+        geometry_column = df._geometry_column_name
+        crs = _encode_crs(df.crs)
+
+        # Convert to a DataFrame so we can convert geometries to WKB while
+        # retaining original column names
+        df = DataFrame(df.copy())
```

Comment:
@jorisvandenbossche since we are modifying the geometry columns but retaining the same names, I think we have to do a deep copy.  Originally I just did `DataFrame(df)` to downcast it, which ended up doing a shallow copy, and resulted in modifying the original (which is why I have tests to make sure we don't modify the original).